### PR TITLE
Anchor contract backward compatibility

### LINKED ISSUE #357: Facilitator init should also support Anchor::getLatestStateRootBlockHeight method to fetch latest block number
M1 anchor contract method name is getLatestStateRootBlockNumber to fetch latest state root block number.
M0 anchor method name is getLatestStateRootBlockHeight.
Facilitator init should support both getLatestStateRootBlockNumber as well as getLatestStateRootBlockHeight.

### PR DESCRIPTION
**Pull request by gulshanvasnani**
PR contains below changes:
1. Making backward compatibility for anchor contract. If the `architecture_layout` value is `MOSAIC_0_14_GEN_1` then chain has Anchor contract of Mosaic-1. If the value for `architecture_ layout` is `MOSAIC_0_14_GEN_0` then the chain has Anchor contract of Mosaic-0 i.e. mosaic-contract-0.10 published npm version.
2. Updated unit test case.

fixes #357

### CODE REVIEW

File: src/m1_facilitator/manifest/Manifest.ts [hunk 1]
```diff
@@ -78,7 +78,10 @@ interface ManifestInfo {
 
 /** Enum of architecture layouts which facilitator supports. */
 export enum ArchitectureLayout {
-  MOSAIC1 = 'MOSAIC1',
+  // For supporting anchor contract in mosaic-1.
+  MOSAIC1_0_14 = 'MOSAIC1_0_14',
```
Comment (deepesh-kn) [inline src/m1_facilitator/manifest/Manifest.ts]: Mosaic_v0.14 can work on both gen0 and gen1 testnets.

Comment (deepesh-kn) [reply src/m1_facilitator/manifest/Manifest.ts]: Why not `MOSAIC_0.14_GEN_0`

File: src/m1_facilitator/manifest/manifest.schema.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
         "architecture_layout": {
           "type": "string",
           "additionalProperties": false,
-          "enum": ["MOSAIC1"]
+          "enum": ["MOSAIC1_0_14", "MOSAIC1_0_10"]
```
Comment (deepesh-kn) [inline src/m1_facilitator/manifest/manifest.schema.json]: May be 
```
"enum": ["MOSAIC_0.14_GEN_0", "MOSAIC_0.14_GEN_1"]
```

File: test/m1_facilitator/SeedDataInitializer/initialize.test.ts [hunk 3]
```diff
@@ -263,4 +274,83 @@ describe('SeedDataInitializer:initialize', () => {
       auxiliaryAnchorAddress,
     );
   });
+
+  it('should initialize seed data when for supporting older testnet', async (): Promise<void> => {
```
Comment (deepesh-kn) [inline test/m1_facilitator/SeedDataInitializer/initialize.test.ts]: What exactly is `older` testnet? 🤔   After few months `Gen1` testnets will also be old.

Comment (gulshanvasnani) [reply test/m1_facilitator/SeedDataInitializer/initialize.test.ts]: Changed it to : `should initialize seed data when facilitator is running on GEN0 testnet`

File: src/m1_facilitator/manifest/manifest.schema.json (hunk 2 above)
Comment (gulshanvasnani) [reply src/m1_facilitator/manifest/manifest.schema.json]: Done

File: src/m1_facilitator/manifest/Manifest.ts (hunk 1 above)
Comment (gulshanvasnani) [reply src/m1_facilitator/manifest/Manifest.ts]: Changed it to :
```
export enum ArchitectureLayout {
  MOSAIC_0_14_GEN_0 = 'MOSAIC_0_14_GEN_0',
  MOSAIC_0_14_GEN_1 = 'MOSAIC_0_14_GEN_1',
}

```

File: src/m1_facilitator/SeedDataInitializer.ts [hunk 4]
```diff
@@ -15,13 +15,15 @@
 import Web3 from 'web3';
 import Mosaic from 'Mosaic';
 import BigNumber from 'bignumber.js';
+import { interacts } from '@openst/mosaic-contracts';
```
Comment (0xsarvesh) [inline src/m1_facilitator/SeedDataInitializer.ts]: should we name this as m0 interacts?

File: src/m1_facilitator/SeedDataInitializer.ts [hunk 5]
```diff
@@ -70,23 +74,39 @@ export default class SeedDataInitializer {
...
-
     const auxiliaryAnchorAddress = await erc20Cogateway.methods.stateRootProvider().call();
-    Logger.debug(`auxiliaryAnchorAddress: ${auxiliaryAnchorAddress}`);
-    const auxiliaryAnchorInstance = Mosaic.interacts.getAnchor(
-      auxiliaryWeb3,
-      auxiliaryAnchorAddress,
-    );
+    let auxiliaryLatestAnchoredStateRootBlockHeight: string;
+    let originLatestAnchoredStateRootBlockHeight: string;
+    if (ArchitectureLayout.MOSAIC_0_14_GEN_1 === architectureLayout) {
```
Comment (0xsarvesh) [inline src/m1_facilitator/SeedDataInitializer.ts:80]: Can you add logs that show which part of if/else is triggered?

**Review by 0xsarvesh — COMMENTED** [review]
Overall looks good 👍  

2 minor comments

File: src/m1_facilitator/SeedDataInitializer.ts (hunk 5 above)
Comment (gulshanvasnani) [reply src/m1_facilitator/SeedDataInitializer.ts:80]: Done

File: src/m1_facilitator/SeedDataInitializer.ts (hunk 4 above)
Comment (gulshanvasnani) [reply src/m1_facilitator/SeedDataInitializer.ts]: Done

**Review by 0xsarvesh — APPROVED** [review]
LGTM 👍

**Pull request merged**